tests/AutoReview/FixerTest.php

```diff
@@ -128,8 +130,25 @@ public function testFixerDefinitions(FixerInterface $fixer)
 
             $options = $fixer->getConfigurationDefinition()->getOptions();
 
+            /** @var FixerOptionInterface $option */
             foreach ($options as $option) {
                 $this->assertRegExp('/^[a-z_]*$/', $option->getName(), sprintf('[%s] Option %s is not snake_case.', $fixerName, $option->getName()));
+
+                $types = $option->getAllowedTypes();
+                $values = $option->getAllowedValues();
+
+                $this->assertFalse(
+                    (null === $types || 0 === count($types)) && (null === $values || 0 === count($values)),
+                    sprintf('[%s] Option %s has no allowed types and values set.', $fixerName, $option->getName())
+                );
+
+                if (null !== $values) {
+                    foreach ($values as $value) {
+                        if ($value instanceof AllowedValueSubset) {
+                            $this->assertContains('array', $types, sprintf('[%s] Option %s has allowed value of type "AllowedValueSubset" but not allowed type "array".', $fixerName, $option->getName()));
```

Comment:
can you show an example for what definition this assertion would fail?
(i'm not fully getting it)

Reply:
```php
        return new FixerConfigurationResolver([
            (new FixerOptionBuilder('elements', sprintf('List of classy elements; \'%s\'.', implode("', '", $types))))
                // ->setAllowedTypes(['array']) this would fail, because `array` is not set as allow type
                ->setAllowedValues([new AllowedValueSubset($types)])
                ->setDefault(['const', 'method', 'property'])
                ->getOption(),
        ]);
```

Reply:
thinking of it, we could also do it the other way around; if the allowed values are defined, there is no need to define the types... hmmm

Reply:
yeah, let us not mix types with values. only one shall be set.

:+1:

Reply:
ultimately (3.0 to not introduce bc breaks?), would be nice to validate in runtime, not on tests run